include/UIMenu.h

```diff
@@ -36,99 +37,104 @@
 class UIMenu;
 class Page;
 
-//typedef void(*MenuCallback)(UIMenu*);
+// typedef void(*MenuCallback)(UIMenu*);
 typedef std::function<void(UIMenu*)> MenuCallback;
 typedef std::function<void(UIMenu*, void*)> ParameterizedMenuCallback;
 typedef std::function<void(UIMenu*, int)> IParameterizedMenuCallback;
+typedef std::function<void(UIMenu*, void*, int)> PIParameterizedMenuCallback;
 
 enum AutocleanMethod {
-  AUTOCLEAN_OFF,
-  AUTOCLEAN_FREE,
-  AUTOCLEAN_DELETE,
+    AUTOCLEAN_OFF,
+    AUTOCLEAN_FREE,
+    AUTOCLEAN_DELETE,
```

Comment:
This is actually a noop, and what we should do instead is pass in a cleanup function that gets a ptr and deals with it. Delete on void* doesn't know which destructor to call.